src/gui/executionlog.cpp

```diff
@@ -35,18 +35,22 @@
 #include <QPalette>
 #include "executionlog.h"
 #include "ui_executionlog.h"
-#include "base/logger.h"
+#include "base/preferences.h"
 #include "guiiconprovider.h"
 #include "loglistwidget.h"
 
 ExecutionLog::ExecutionLog(QWidget *parent)
     : QWidget(parent)
     , ui(new Ui::ExecutionLog)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES))
+    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, true, true, true, true))
```

Comment:
>By and large, this class should not contain any logic associated with handling log messages (and only the initialization/configuration of interface elements). This should be moved to the appropriate widget code

Forget it if you want. Actually, I understand why you're using this proxy code. You just want to have a simple widget (LogListWidget) that would operate only with some abstract string messages and their categories. I just somewhat confused by the fact that categories aren't used to working with peers... Nothing. Just do the default values for constructor parameters.